src/Enforcers/ValidationEnforcer.php

```diff
@@ -0,0 +1,177 @@
+     *
+     * @param string className - The name of the class for which the validation metadata
+     * is to be created.
+     * @return array See above description.
+     */
+    public function getValidationMetadata(string $className): array
+    {
+        $validators = [
+            "int"      => new IntValidator(),
+            "string"   => new StringValidator(),
+            "bool"     => new BoolValidator(),
+            "DateTime" => new DateValidator(),
+            "notNull"  => new NotNullValidator(),
+        ];
+
+        $metadata = [];
+
+        $ref   = new ReflectionClass($className);
+        $props = $ref->getProperties();
+
+        //For each property defined on the class.
+        foreach($props as $prop)
+        {
+            $propName    = $prop->getName();
+            $propType    = $prop->getType()->getName();
+            $propAttrs   = $prop->getAttributes(Validator::class, ReflectionAttribute::IS_INSTANCEOF);
+            $propCanNull = $prop->getType()->allowsNull();
+
+            $metadata[$propName] = ['reflectionProperty' => $prop, 'validators' => []];
+            $validator = array_key_exists($propType, $validators) ? $validators[$propType] : null;
+
+            if (!is_null($validator))
+            {
+                $metadata[$propName]['validators'][] = $validator;
+            }
+
+            if (!$propCanNull)
+            {
+                $metadata[$propName]['validators'][] = $validators['notNull'];
+            }
+
+            //If the property has any validation attributes make sure they are included.
+            foreach($propAttrs as $attr)
+            {
+                $metadata[$propName]['validators'][] = $attr->newInstance();
+            }
+        }
+
+        return $metadata;
+    }
+
+    /**
+     * Returns true if the property is defined on the object.
+     *
+     * @param array|object $object - Either an object or an array on which to determine
+     * if the property is defined.
+     * @param string $property - The property being looked for.
+     * @return bool Returns true if the property is defined on the object.
+     */
+    public function propIsDefined(mixed $object, string $property): bool
+    {
+        $arr = is_array($object) ? $object : (array) $object;
+
+        return array_key_exists($property, $arr);
+    }
+
+    /**
+     * Returns true if the property is defined as null on the object.
+     *
+     * @param array|object $object - Either an object or an array on which to determine
+     * if the property is defined as null.
+     * @param string $property - The property being looked at.
+     * @return bool Returns true if the property is defined as null on the object.
+     */
+    public function propIsNull(mixed $object, string $property): bool
+    {
+        $arr = is_array($object) ? $object : (array) $object;
+
+        return $this->propIsDefined($arr, $property) && is_null($arr["$property"]);
```

Comment:
Not sure if we need the double quotes around `$property` in  `$arr["$property"]`.